src/autorest-core/language-service/language-service.ts

```diff
@@ -0,0 +1,653 @@
+          getText: () => txt,
+          positionAt: (offset: number) => <Position>{},
+          offsetAt: (position: Position) => 0,
+          lineCount: 1
+        });
+      }
+    }
+  }
+  // IFileSystem Implementation
+  public async EnumerateFileUris(folderUri: string): Promise<Array<string>> {
+    if (folderUri && folderUri.startsWith("file:")) {
+      const folderPath = FileUriToPath(folderUri);
+      if (await isDirectory(folderPath)) {
+        const items = await readdir(folderPath);
+        return From<string>(items).Where(each => AutoRest.IsConfigurationExtension(GetExtension(each))).Select(each => ResolveUri(folderUri, each)).ToArray();
+      }
+    }
+
+    return [];
+  }
+
+  public async ReadFile(fileUri: string): Promise<string> {
+    const doc = this.get(fileUri) || this.virtualFile.get(fileUri);
+    try {
+      if (doc) {
+        return doc.getText();
+      }
+
+      const content = await readFile(FileUriToPath(fileUri));
+
+      return content;
+    } catch {
+    }
+    return "";
+  }
+
+  private async process(configurationUrl: string) {
+    const result = this.results.get(configurationUrl) || new Result(this, configurationUrl);
+    this.results.set(configurationUrl, result);
+
+    // ensure that we are no longer processing a previous run.
+    await result.cancel();
+
+    // process the files.
+    await result.process();
+  }
+
+  private async getConfiguration(documentUri: string): Promise<string> {
+    // let folder = ResolveUri(documentUri, ".");
+    let configFiles = await Configuration.DetectConfigurationFiles(this, documentUri, undefined, true);
+
+    // is the document a config file?
+    if (configFiles.length === 1 && configFiles[0] == documentUri) {
+      return documentUri;
+    }
+
+    // is there a config file that contains the document as an input?
+    for (const configFile of configFiles) {
+      const a = new AutoRest(this, configFile);
+      const inputs = (await a.view).InputFileUris
+      for (const input of inputs) {
+        if (input === documentUri || decodeURIComponent(input) == decodeURIComponent(documentUri)) {
+          return configFile;
+        }
+      }
+    }
+
+    // didn't find a match, let's make a dummy one.
+    const configFile = `${documentUri}/readme.md`;
+    if (!this.virtualFile.get(configFile)) {
+      this.virtualFile.set(configFile, {
+        uri: configFile,
+        languageId: "markdown",
+        version: 1,
+        getText: () => "#Fake config file \n> see https://aka.ms/autorest \n``` yaml \ninput-file: \n - " + documentUri,
+        positionAt: (offset: number) => <Position>{},
+        offsetAt: (position: Position) => 0,
+        lineCount: 1
+      });
+    }
+
+    return configFile;
+  }
+
+  private async onDocumentChanged(document: TextDocument) {
+    this.debug(`onDocumentChanged: ${document.uri}`);
+
+    if (AutoRest.IsSwaggerExtension(document.languageId) && await AutoRest.IsSwaggerFile(document.getText())) {
```

Comment:
in both of these cases from the vs code extension point of view, we'd like to have azure-validator: true set by default and openapi-type: arm. If they are in the values are set in the configuration already, we should take those, but if not we want to set them. Could we set configuration options here? Can we have these values also being picked up for the case where we're creating a dummy readme.md file?